app/Models/Game.php

```diff
@@ -507,6 +488,11 @@ public function achievementSetClaims(): HasMany
         return $this->hasMany(AchievementSetClaim::class, 'game_id');
     }
 
+    public function parentGame(): ?Game
+    {
+        return $this->parentGameId ? Game::find($this->parentGameId) : null;
```

Comment:
Correct me if I'm wrong, but `parentGameId` will only be evaluated once and stored in the model. Right?

Reply:
I'm 99% sure it'll query the DB every time it's called, even on the same model. This sort of caching only works for actual relationships.

I think you have two options:
* Convert it to a relationship, or
* Use Laravel's built-in `once()` helper, [docs](https://laravel.com/docs/11.x/helpers#method-once).

```php
// untested psuedocode
return once(fn () => $this->parentGameId ? Game::find().......
```

Reply:
Confirmed :(
```
before parentGame()
in parentGameIdAttribute
in parentGameIdAttribute
after parentGame()
```

Reply:
This `return` should probably also be wrapped in `once()` (similar to the pseudocode example above).